File: test/VbNet.LanguageServer.Tests/Services/FormattingServiceTests.cs
using Microsoft.Extensions.Logging.Abstractions;
using VbNet.LanguageServer.Protocol;
using VbNet.LanguageServer.Services;
using VbNet.LanguageServer.Workspace;
using Xunit;

namespace VbNet.LanguageServer.Tests.Services;

/// <summary>
/// Unit tests for FormattingService.
/// </summary>
public class FormattingServiceTests
{
    private readonly WorkspaceManager _workspaceManager;
    private readonly DocumentManager _documentManager;
    private readonly FormattingService _formattingService;

    public FormattingServiceTests()
    {
        _workspaceManager = new WorkspaceManager(NullLogger<WorkspaceManager>.Instance);
        _documentManager = new DocumentManager(_workspaceManager, NullLogger<DocumentManager>.Instance);
        _formattingService = new FormattingService(
            _workspaceManager,
            _documentManager,
            NullLogger<FormattingService>.Instance);

        _workspaceManager.Initialize();
    }

    [Fact]
    public async Task FormatDocumentAsync_NoDocument_ReturnsEmpty()
    {
        var @params = new DocumentFormattingParams
        {
            TextDocument = new TextDocumentIdentifier { Uri = "file:///nonexistent.vb" }
        };

        var result = await _formattingService.FormatDocumentAsync(@params, CancellationToken.None);

        Assert.NotNull(result);
        Assert.Empty(result);
    }

    [Fact]
    public async Task FormatRangeAsync_NoDocument_ReturnsEmpty()
    {
        var @params = new DocumentRangeFormattingParams
        {
            TextDocument = new TextDocumentIdentifier { Uri = "file:///nonexistent.vb" },
            Range = new VbNet.LanguageServer.Protocol.Range
            {
                Start = new Position(0, 0),
                End = new Position(0, 1)
            }
        };

        var result = await _formattingService.FormatRangeAsync(@params, CancellationToken.None);

        Assert.NotNull(result);
        Assert.Empty(result);
    }
}

File: test/VbNet.LanguageServer.Tests.Vb/Services/FormattingServiceTests.vb
Imports System.Threading
Imports System.Threading.Tasks
Imports Microsoft.Extensions.Logging.Abstractions
Imports VbNet.LanguageServer.Protocol
Imports VbNet.LanguageServer.Services
Imports VbNet.LanguageServer.Workspace
Imports Xunit

Namespace VbNet.LanguageServer.Tests.Services

    ''' <summary>
    ''' Unit tests for FormattingService.
    ''' </summary>
    Public Class FormattingServiceTests
        Private ReadOnly _workspaceManager As WorkspaceManager
        Private ReadOnly _documentManager As DocumentManager
        Private ReadOnly _formattingService As FormattingService

        Public Sub New()
            _workspaceManager = New WorkspaceManager(NullLogger(Of WorkspaceManager).Instance)
            _documentManager = New DocumentManager(_workspaceManager, NullLogger(Of DocumentManager).Instance)
            _formattingService = New FormattingService(
                _workspaceManager,
                _documentManager,
                NullLogger(Of FormattingService).Instance)

            _workspaceManager.Initialize()
        End Sub

        <Fact>
        Public Async Function FormatDocumentAsync_NoDocument_ReturnsEmpty() As Task
            Dim request = New DocumentFormattingParams With {
                .TextDocument = New TextDocumentIdentifier With {.Uri = "file:///nonexistent.vb"}
            }

            Dim result = Await _formattingService.FormatDocumentAsync(request, CancellationToken.None)

            Assert.NotNull(result)
            Assert.Empty(result)
        End Function

        <Fact>
        Public Async Function FormatRangeAsync_NoDocument_ReturnsEmpty() As Task
            Dim request = New DocumentRangeFormattingParams With {
                .TextDocument = New TextDocumentIdentifier With {.Uri = "file:///nonexistent.vb"},
                .Range = New Global.VbNet.LanguageServer.Protocol.Range With {
                    .Start = New Position(0, 0),
                    .End = New Position(0, 1)
                }
            }

            Dim result = Await _formattingService.FormatRangeAsync(request, CancellationToken.None)

            Assert.NotNull(result)
            Assert.Empty(result)
        End Function
    End Class

End Namespace
